libkbfs/folder_branch_ops.go

```diff
@@ -3801,21 +3830,39 @@ func (fbo *folderBranchOps) syncAllLocked(
 			continue
 		}
 
-		// For rename and setattr ops, the target file will have a
-		// dirty entry, but may not have any outstanding writes, so it
-		// needs to be cleaned up manually.
+		// For create, rename and setattr ops, the target will have a
+		// dirty entry, but may not have any outstanding operations on
+		// it, so it needs to be cleaned up manually.
 		defer func() {
 			if err != nil {
 				return
 			}
-			fbo.blocks.ClearCachedRef(lState, ref)
+			wasCleared := fbo.blocks.ClearCachedRef(lState, ref)
+			if wasCleared {
+				node := fbo.nodeCache.Get(ref)
+				if node != nil {
+					fbo.status.rmDirtyNode(node)
+				}
+			}
 		}()
 	}
 
-	fileBlocks := make(fileBlockMap)
 	var blocksToRemove []BlockPointer
 	var afterUpdateFns []func() error
 
+	afterUpdateFns = append(afterUpdateFns, func() error {
```

Comment:
This seems like it's going to do a lot of dynamic dispatch of closures. Not sure of a better way right now though. Maybe a `TODO` to revisit?